Makefile

```diff
@@ -17,4 +17,4 @@ deps:
 	go get github.com/golangci/golangci-lint/cmd/golangci-lint
 
 generate-mocks:
-	docker run --rm -v ${PWD}:/src -w /src golang:1.20-alpine ./hack/generate-mocks.sh
+	mockery
```

Comment:
@flavio generating the mocks with go generate requires the mockery binary to be installed. We are using the `.mockery.yaml` config file instead.

Reply:
@flavio `go generate` requires the binary to be installed, as it just runs the command in the comment. I prefer having the `.mockery.yaml` configuration since it is easier to read than a golang comment.

Reply:
I'm ok with the approach. Also, the Makefile is a bit of a mess.. no phony targets and so. But not needed to fix in this PR.

Reply:
@viccuad could you open an issue about the Makefile?